### PR TITLE
Enable DML on Windows and CUDA on Linux for Node.js binding

### PR DESCRIPTION
**Pull request by jchen351**
This pull request includes modifications to the `c-api-cpu.yml` Azure Pipelines configuration file. The changes mainly revolve around the Node.js packaging stage and the handling of Node.js artifacts. The most significant changes include renaming the Node.js packaging stage, adding a new dependency to the stage, changing artifact names, adding a new script to list Node.js artifacts, and updating the source folder for copying NuGet binaries.

Changes in Node.js packaging:

* [`tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml`](diffhunk://#diff-00815920cc190d10fdebceac0c3a4b8a59e408684ae38177dfe7f96cae276c59L503-R508): Renamed the Node.js packaging stage from `Nodejs_Packaging_CPU` to `Nodejs_Packaging` and added `Windows_CI_GPU_DML_Dev` as a new dependency to the stage.

Changes in handling of Node.js artifacts:

* [`tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml`](diffhunk://#diff-00815920cc190d10fdebceac0c3a4b8a59e408684ae38177dfe7f96cae276c59L568-R569): Changed the artifact name from `drop-onnxruntime-nodejs-win-x64` to `drop-onnxruntime-nodejs-win-x64-dml` in the task to download pipeline artifacts for Windows x64.
* [`tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml`](diffhunk://#diff-00815920cc190d10fdebceac0c3a4b8a59e408684ae38177dfe7f96cae276c59R595-R598): Added a new script to list Node.js artifacts from the directory `$(Build.BinariesDirectory)/nodejs-artifacts/win32/x64/`.
* [`tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml`](diffhunk://#diff-00815920cc190d10fdebceac0c3a4b8a59e408684ae38177dfe7f96cae276c59L635-R640): Updated the source folder from `$(Build.BinariesDirectory)\RelWithDebInfo\RelWithDebInfo\nuget-artifacts\onnxruntime-win-x64\lib` to `$(Build.BinariesDirectory)\nodejs-artifacts\win32\x64` in the task to copy NuGet binaries to the directory `$(Build.SourcesDirectory)\js\node\bin\napi-v3\win32\x64`.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml [hunk 1]
```diff
@@ -500,11 +500,12 @@ stages:
       displayName: 'Clean Agent Directories'
       condition: always()
 
-- stage: Nodejs_Packaging_CPU
+- stage: Nodejs_Packaging
   dependsOn:
   - Linux_C_API_Packaging_CPU
   - MacOS_C_API_Package_Publish
   - Windows_Packaging_CPU_x64_${{ parameters.BuildVariant }}
```
Comment (fs-eire) [inline tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml:507]: I think `Windows_Packaging_CPU_x64_${{ parameters.BuildVariant }}` is no longer need?

File: tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml [hunk 2]
```diff
@@ -591,6 +592,10 @@ stages:
       inputs:
         artifactName: 'drop-onnxruntime-nodejs-linux-x64'
         targetPath: '$(Build.BinariesDirectory)/nodejs-artifacts/linux/x64/'
+    - script: |
+        dir
+      workingDirectory: '$(Build.BinariesDirectory)/nodejs-artifacts/win32/x64/'
+      displayName: 'List Nodejs artifacts from $(Build.BinariesDirectory)/nodejs-artifacts/win32/x64/'
```
Comment (fs-eire) [inline tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml:598]: is this step only for debug purpose?

**Comment by snnn** [comment]
How is it going?

**Comment by fs-eire** [comment]
I updated this PR and added cuda on linux as well. Queuing build on CI and waiting for the result.

**Comment by fs-eire** [comment]
Now it should work with the CI build. @snnn @jchen351 please help to take a look.

**Review by snnn — APPROVED** [review]

**Comment by snnn** [comment]
"Zip-Nuget-Java-Nodejs Packaging Pipeline" has a test that ran timeout, but your changes look good so we can merge this PR first then solve it separately.

**Pull request merged**